app/api/views/teams.py

```diff
@@ -0,0 +1,108 @@
+import rollbar
+from flask import jsonify, request
+from flask_login import current_user, login_required
+
+from app.api import api
+from app.api.business import team_business
+from app.api.business.errors import NotFoundError, TeamError, ValidationError, UnauthorisedError
+from app.api.helpers import abort, forbidden, get_email_domain, role_required, not_found
+from app.api.services import team_member_service, users
+from ...utils import get_json_from_request
+
+
+@api.route('/teams', methods=['GET'])
+@login_required
+@role_required('buyer')
+def get_teams_overview():
+    team = team_business.get_teams_overview()
+
+    return jsonify(team)
+
+
+@api.route('/people', methods=['GET'])
+@login_required
+@role_required('buyer')
+def get_people_overview():
+    team = team_business.get_people_overview()
+
+    return jsonify(team)
+
+
+@api.route('/team/create', methods=['POST'])
+@login_required
+@role_required('buyer')
+def create_team():
+    try:
+        team = team_business.create_team()
+    except TeamError as e:
+        abort(e.message)
+    except Exception as e:
+        rollbar.report_exc_info()
+
+    return jsonify(team)
+
+
+@api.route('/team/<int:team_id>', methods=["GET"])
+@login_required
+@role_required('buyer')
+def get_team(team_id):
+    team = None
+    try:
+        team = team_business.get_team(team_id)
+    except NotFoundError as e:
+        return not_found(e.message)
+    except UnauthorisedError as e:
+        return forbidden(e.message)
+
+    return jsonify(team)
+
+
+@api.route('/team/<int:team_id>/update', methods=['POST'])
+@login_required
+@role_required('buyer')
+def update_team(team_id):
+    data = get_json_from_request()
+    try:
+        team = team_business.update_team(team_id, data)
```

Comment:
As a buyer whose email is domain X, I'm able to add a buyer user as a team member whose domain is not X. As the request only needs a user ID to add a team member, and the response to this request is enriched with email address and name of that user id, a buyer account could use this to iterate our buyer user base.

Reply:
fixed in validator